Sources/TensorFlow/Operators/Basic.swift

```diff
@@ -1325,42 +1325,75 @@ extension Tensor.IndexPath {
 //===------------------------------------------------------------------------------------------===//
 
 extension Tensor {
-  /// Returns `true` if the given axis is in the range `[-rank, rank)`.
+  /// Returns `true` iff `axis` can refer to an axis of `self`.
   @usableFromInline
-  internal func isAxisInRange<T: BinaryInteger>(_ axis: T) -> Bool {
+  internal func isValid<T: BinaryInteger>(axis: T) -> Bool {
     let axis = Int(axis)
     return axis >= -rank && axis < rank
   }
 
-  /// Returns `true` if the given scalar tensor is in the range `[-rank, rank)`.
+  /// Returns `true` iff all scalars can refer to an axis of `self`.
   @usableFromInline
-  internal func isAxisInRange(
-    _ axis: Tensor<Int32>,
+  internal func areValid<T: BinaryInteger>(axes: [T]) -> Bool {
+    return axes.allSatisfy { isValid(axis: $0) }
+  }
+
+  /// Returns `true` iff all scalars can refer to an axis of `self`.
+  @usableFromInline
+  internal func areValid(
+    axes: Tensor<Int32>,
     file: StaticString = #file,
     line: UInt = #line
   ) -> Bool {
-    precondition(axis.rank == 0, "Axis must have rank 0.", file: file, line: line)
-    return areAxesInRange(axis.scalars)
+    precondition(
+      axes.rank < 2,
+      "Axes must have rank 0 or rank 1; axes has rank \(axes.rank) with values \(axes.scalars).",
+      file: file,
+      line: line)
+    return areValid(axes: axes.scalars)
   }
 
-  /// Returns `true` if all given axes are in the range `[-rank, rank)`.
+  /// Ensures the provided axes refer to axis of `self` (are in range `-rank..<rank`).
```

Comment:
“Ensures” doesn't have a very crisp meaning.  To me it sounds like it's maybe going to modify `axes` so its elements are in range.  
```suggestion
  /// Checks that each element of `axes` denotes an element of
  /// `self`, and stops the program with a diagnostic otherwise. 
```